### ISSUE TITLE
Publish data as NPM package

### ISSUE THREAD

**Issue by YusukeHirao** [accessibilitysupported/a11ysupport.io#92]
Hi!

I am write the HTML linter named [markuplint](https://www.npmjs.com/package/markuplint) that published to NPM.

A linter can check attribute and that value in HTML specs already. I will add ARIA attribute availabilities on screen readers to a linter. Because of that, I hope to use your data.

Would you publish data as NPM package like a [canuise-db](https://www.npmjs.com/package/caniuse-db)?

**Comment by mfairchild365** [comment]
Hi @YusukeHirao - I don't have any plans to implement an npm package like this yet. The project is still evolving and I think it needs to mature a bit before we can approach creating a package like that. I do think it is a wonderful idea, and I'd love to tackle it in the future, so I'll leave this issue open.

**Comment by YusukeHirao** [comment]
Thank you for your reply.  I'll be waiting patiently.

**Comment by schne324** [comment]
It looks like the build script handles everything needed to publish the data to npm.

You'd just have to add a `prepublishOnly` script to `package.json` to run the build before publishing (optionally, [package.json's "files" field](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#files) could be utilized to restrict what is published to only the `build/` directory).

---

<details>
  <summary>what would be published to npm</summary>

```
├── command_matrix.json
├── features.json
├── support_points.json
├── tech
│   ├── aria
│   │   ├── alert_role.json
│   │   ├── alertdialog_role.json
│   │   ├── application_role.json
│   │   ├── aria-activedescendant_attribute.json
│   │   ├── aria-atomic_attribute.json
│   │   ├── aria-autocomplete_attribute.json
│   │   ├── aria-busy_attribute.json
│   │   ├── aria-checked_attribute.json
│   │   ├── aria-colcount_attribute.json
│   │   ├── aria-colindex_attribute.json
│   │   ├── aria-colspan_attribute.json
│   │   ├── aria-controls_attribute.json
│   │   ├── aria-current_attribute.json
│   │   ├── aria-describedby_attribute.json
│   │   ├── aria-details_attribute.json
│   │   ├── aria-disabled_attribute.json
│   │   ├── aria-dropeffect_attribute.json
│   │   ├── aria-errormessage_attribute.json
│   │   ├── aria-expanded_attribute.json
│   │   ├── aria-flowto_attribute.json
│   │   ├── aria-grabbed_attribute.json
│   │   ├── aria-haspopup_attribute.json
│   │   ├── aria-hidden_attribute.json
│   │   ├── aria-invalid_attribute.json
│   │   ├── aria-keyshortcuts_attribute.json
│   │   ├── aria-label_attribute.json
│   │   ├── aria-labelledby_attribute.json
│   │   ├── aria-level_attribute.json
│   │   ├── aria-live_attribute.json
│   │   ├── aria-modal_attribute.json
│   │   ├── aria-multiline_attribute.json
│   │   ├── aria-multiselectable_attribute.json
│   │   ├── aria-orientation_attribute.json
│   │   ├── aria-owns_attribute.json
│   │   ├── aria-placeholder_attribute.json
│   │   ├── aria-posinset_attribute.json
│   │   ├── aria-pressed_attribute.json
│   │   ├── aria-readonly_attribute.json
│   │   ├── aria-relevant_attribute.json
│   │   ├── aria-required_attribute.json
│   │   ├── aria-roledescription_attribute.json
│   │   ├── aria-rowcount_attribute.json
│   │   ├── aria-rowindex_attribute.json
│   │   ├── aria-rowspan_attribute.json
│   │   ├── aria-selected_attribute.json
│   │   ├── aria-setsize_attribute.json
│   │   ├── aria-sort_attribute.json
│   │   ├── aria-valuemax_attribute.json
│   │   ├── aria-valuemin_attribute.json
│   │   ├── aria-valuenow_attribute.json
│   │   ├── aria-valuetext_attribute.json
│   │   ├── article_role.json
│   │   ├── banner_role.json
│   │   ├── button_role.json
│   │   ├── cell_role.json
│   │   ├── checkbox_role.json
│   │   ├── columnheader_role.json
│   │   ├── combobox_role.json
│   │   ├── command_role.json
│   │   ├── complementary_role.json
│   │   ├── composite_role.json
│   │   ├── contentinfo_role.json
│   │   ├── definition_role.json
│   │   ├── dialog_role.json
│   │   ├── directory_role.json
│   │   ├── document_role.json
│   │   ├── feed_role.json
│   │   ├── figure_role.json
│   │   ├── form_role.json
│   │   ├── grid_role.json
│   │   ├── gridcell_role.json
│   │   ├── group_role.json
│   │   ├── heading_role.json
│   │   ├── img_role.json
│   │   ├── input_role.json
│   │   ├── landmark_role.json
│   │   ├── link_role.json
│   │   ├── list_role.json
│   │   ├── listbox_role.json
│   │   ├── listitem_role.json
│   │   ├── log_role.json
│   │   ├── main_role.json
│   │   ├── marquee_role.json
│   │   ├── math_role.json
│   │   ├── menu_role.json
│   │   ├── menubar_role.json
│   │   ├── menuitem_role.json
│   │   ├── menuitemcheckbox_role.json
│   │   ├── menuitemradio_role.json
│   │   ├── navigation_role.json
│   │   ├── none_role.json
│   │   ├── note_role.json
│   │   ├── option_role.json
│   │   ├── presentation_role.json
│   │   ├── progressbar_role.json
│   │   ├── radio_role.json
│   │   ├── radiogroup_role.json
│   │   ├── range_role.json
│   │   ├── region_role.json
│   │   ├── roletype_role.json
│   │   ├── row_role.json
│   │   ├── rowgroup_role.json
│   │   ├── rowheader_role.json
│   │   ├── scrollbar_role.json
│   │   ├── search_role.json
│   │   ├── searchbox_role.json
│   │   ├── section_role.json
│   │   ├── sectionhead_role.json
│   │   ├── select_role.json
│   │   ├── separator_role.json
│   │   ├── slider_role.json
│   │   ├── spinbutton_role.json
│   │   ├── status_role.json
│   │   ├── structure_role.json
│   │   ├── switch_role.json
│   │   ├── tab_role.json
│   │   ├── table_role.json
│   │   ├── tablist_role.json
│   │   ├── tabpanel_role.json
│   │   ├── term_role.json
│   │   ├── textbox_role.json
│   │   ├── timer_role.json
│   │   ├── toolbar_role.json
│   │   ├── tooltip_role.json
│   │   ├── tree_role.json
│   │   ├── treegrid_role.json
│   │   ├── treeitem_role.json
│   │   ├── widget_role.json
│   │   └── window_role.json
│   ├── css
│   │   ├── generated_content.json
│   │   └── generated_content_alt.json
│   ├── html
│   │   ├── a(href)_element.json
│   │   ├── a_element.json
│   │   ├── button_element.json
│   │   ├── canvas_element.json
│   │   ├── datalist_element.json
│   │   ├── dd_element.json
│   │   ├── details_element.json
│   │   ├── dialog_element.json
│   │   ├── disabled_attribute.json
│   │   ├── div_element.json
│   │   ├── dl_element.json
│   │   ├── dt_element.json
│   │   ├── fieldset_element.json
│   │   ├── figcaption_element.json
│   │   ├── figure_element.json
│   │   ├── h1-6_elements.json
│   │   ├── headers_attribute.json
│   │   ├── img_element.json
│   │   ├── input(type-button)_element.json
│   │   ├── input(type-checkbox)_element.json
│   │   ├── input(type-color)_element.json
│   │   ├── input(type-date)_element.json
│   │   ├── input(type-datetime-local)_element.json
│   │   ├── input(type-email)_element.json
│   │   ├── input(type-file)_element.json
│   │   ├── input(type-hidden)_element.json
│   │   ├── input(type-image)_element.json
│   │   ├── input(type-month)_element.json
│   │   ├── input(type-number)_element.json
│   │   ├── input(type-password)_element.json
│   │   ├── input(type-radio)_element.json
│   │   ├── input(type-range)_element.json
│   │   ├── input(type-reset)_element.json
│   │   ├── input(type-search)_element.json
│   │   ├── input(type-submit)_element.json
│   │   ├── input(type-tel)_element.json
│   │   ├── input(type-text)_element.json
│   │   ├── input(type-time)_element.json
│   │   ├── input(type-url)_element.json
│   │   ├── input(type-week)_element.json
│   │   ├── label_element.json
│   │   ├── lang_attribute.json
│   │   ├── legend_element.json
│   │   ├── li_element.json
│   │   ├── list_attribute.json
│   │   ├── max_attribute.json
│   │   ├── maxlength_attribute.json
│   │   ├── min_attribute.json
│   │   ├── ol_element.json
│   │   ├── optgroup_element.json
│   │   ├── option_element.json
│   │   ├── p_element.json
│   │   ├── required_attribute.json
│   │   ├── role_attribute.json
│   │   ├── scope_attribute.json
│   │   ├── section_element.json
│   │   ├── select_element.json
│   │   ├── span_element.json
│   │   ├── summary_element.json
│   │   ├── svg_element.json
│   │   ├── table_element.json
│   │   ├── td_element.json
│   │   ├── textarea_element.json
│   │   ├── th_element.json
│   │   └── ul_element.json
│   └── svg
│       └── title_element.json
├── tech.json
├── test_map.json
├── tests
│   ├── apg
│   │   ├── aria-1-2-combobox-with-list-autocomplete-example.json
│   │   ├── checkbox-two-state.json
│   │   ├── menu-button-actions-active-descendant.json
│   │   └── modal-dialog-example.json
│   ├── aria_alertdialog_document_mode.json
│   ├── aria_dialog_document_mode.json
│   ├── aria_gridcell(aria-selected).json
│   ├── aria_roledescription_with_html_button.json
│   ├── aria_roledescription_with_html_section.json
│   ├── headings.json
│   ├── html_button(type-button--aria-disabled-true).json
│   ├── html_button(type-button--disabled).json
│   ├── html_label_element_explicit.json
│   ├── html_label_element_implicit.json
│   ├── html_option_element_with_lang.json
│   ├── html_p_element_with_lang.json
│   ├── html_section(aria-label).json
│   ├── html_section.json
│   ├── html_svg_inline_with_title.json
│   ├── html_td_element_with_th(scope-row).json
│   └── tech
│       ├── aria
│       │   ├── alert-role-named.json
│       │   ├── alert-role-unnamed.json
│       │   ├── aria-atomic-divs.json
│       │   ├── aria-atomic-spans.json
│       │   ├── aria-autocomplete.json
│       │   ├── aria-busy.json
│       │   ├── aria-controls.json
│       │   ├── aria-current.json
│       │   ├── aria-describedby-with-role-alert.json
│       │   ├── aria-describedby.json
│       │   ├── aria-details.json
│       │   ├── aria-errormessage-with-aria-invalid-true.json
│       │   ├── aria-expanded.json
│       │   ├── aria-flowto.json
│       │   ├── aria-haspopup-attribute.json
│       │   ├── aria-invalid.json
│       │   ├── aria-label.json
│       │   ├── aria-labelledby-one-ref.json
│       │   ├── aria-labelledby-two-ref.json
│       │   ├── aria-live.json
│       │   ├── aria-owns-multiple.json
│       │   ├── aria-owns-single.json
│       │   ├── aria-pressed.json
│       │   ├── aria-relevant.json
│       │   ├── aria-required-on-html-radio-buttons.json
│       │   ├── aria-required-radiogroup.json
│       │   ├── aria-required.json
│       │   ├── aria-sort.json
│       │   ├── grid.json
│       │   ├── heading-role-no-level.json
│       │   ├── log-role-named.json
│       │   ├── log-role-unnamed.json
│       │   ├── role-button-name-from-inner.json
│       │   ├── role-button-name-from-label.json
│       │   ├── role-group-unnamed.json
│       │   ├── role-link-name-from-inner.json
│       │   ├── role-link-name-from-label.json
│       │   ├── status-role-named.json
│       │   └── status-role-unnamed.json
│       ├── css
│       │   ├── css_generated_content_alt.json
│       │   ├── css_generated_content_button.json
│       │   ├── css_generated_content_in_form_label.json
│       │   ├── css_generated_content_link.json
│       │   └── css_generated_content_span.json
│       └── html
│           ├── button-name-change.json
│           ├── buttons.json
│           ├── datalist.json
│           ├── details-summary.json
│           ├── dl-basic.json
│           ├── dl-groups-wrapped.json
│           ├── headers-attribute-multi-level.json
│           ├── headers-attribute.json
│           ├── html_figure
│           │   ├── test_1.json
│           │   ├── test_2.json
│           │   └── test_3.json
│           ├── input
│           │   ├── input-checkbox.json
│           │   ├── input-color.json
│           │   ├── input-date.json
│           │   ├── input-datetime-local.json
│           │   ├── input-email.json
│           │   ├── input-file.json
│           │   ├── input-month.json
│           │   ├── input-number.json
│           │   ├── input-password.json
│           │   ├── input-range.json
│           │   ├── input-search.json
│           │   ├── input-tel.json
│           │   ├── input-text.json
│           │   ├── input-time.json
│           │   ├── input-url.json
│           │   └── input-week.json
│           ├── links
│           │   ├── example1.json
│           │   ├── example2.json
│           │   ├── example3.json
│           │   ├── example4.json
│           │   ├── example5.json
│           │   ├── example6.json
│           │   ├── example7.json
│           │   └── example8.json
│           ├── maxlength-attribute.json
│           ├── ol.json
│           ├── required_attribute_radio_group.json
│           ├── role-attribute.json
│           ├── role-vc.json
│           ├── select.json
│           ├── table
│           │   └── example_1.json
│           ├── textarea.json
│           └── ul.json
└── tests.json
```
</details>


```js
// example usage
const alertRoleData = require('a11ysupport/tech/aria/alert_role.json');
```

**Comment by mfairchild365** [comment]
Thanks, @schne324 - that is super helpful. Looking into this further, and the `build` directory is about 100mb. That seems like a rather large NPM package. I haven't tried to publish it, but a quick google search says publishing packages over 10mb can get a bit dicey. 

Do y'all need everything in that directory, or just certain things? Is there an opportunity for me to build a specific file (or files) that are optimized for your use-case?

**Comment by mfairchild365** [comment]
@schne324 - there is some desire to work on this as part of the axe-hackathon. Are there any use-cases for this sort of information re consumption by axe-core or other products? If there are, what data do you need for a MVP? Any way we can coordinate?

@YusukeHirao Happy to hear your thoughts as well. What data would you need for a MVP?

**Comment by schne324** [comment]
> publishing packages over 10mb can get a bit dicey

definitely true for most packages but since we're talking about a data package, I'm not sure that this would be an issue. Consumers of this could import only what they need. In fact, this could actually be used server-side to prevent this from adding bloat to the front-end 😄 !

> Do y'all need everything in that directory?

I think so, as I described above, I think having everything shouldn't be a problem.

> what data do you need for a MVP?

I honestly think a good MVP would be to just publish what we've got in the `build/` directory. I am more than happy to take this on and submit a PR, just let me know 😄
